pkg/sdkserver/localsdk.go

```diff
@@ -617,30 +620,167 @@ func (l *LocalSDKServer) UpdateCounter(ctx context.Context, in *alpha.UpdateCoun
+		tmpCounter := &alpha.Counter{Name: name, Count: counter.Count, Capacity: counter.Capacity}
+		// Removes any fields from the request object that are not included in the FieldMask Paths.
+		fmutils.Filter(in.Counter, in.UpdateMask.Paths)
+		// Removes any fields from the existing gameserver object that are included in the FieldMask Paths.
+		fmutils.Prune(tmpCounter, in.UpdateMask.Paths)
+		// Due due filtering and pruning all gameserver object field(s) contained in the FieldMask are overwritten by the request object field(s).
+		proto.Merge(tmpCounter, in.Counter)
+		// Verify that 0 <= Count >= Capacity
+		if tmpCounter.Count < 0 || tmpCounter.Count > tmpCounter.Capacity {
+			return nil, errors.Errorf("OUT_OF_RANGE. Count must be within range [0,Capacity]. Found Count: %d, Capacity: %d", tmpCounter.Count, tmpCounter.Capacity)
 		}
-		return nil, errors.Errorf("INVALID_ARGUMENT. Field Mask Path(s) %v are invalid for Counter. Use valid %v", in.UpdateMask.GetPaths(), counter.ProtoReflect().Descriptor().Fields())
+		// TODO: Verify that Capacity is within bounds determined by CRD.
+		// Write newly updated Counter to gameserverstatus.
+		l.gs.Status.Counters[name].Count = tmpCounter.Count
+		l.gs.Status.Counters[name].Capacity = tmpCounter.Capacity
+		return &alpha.Counter{Name: name, Count: l.gs.Status.Counters[name].Count, Capacity: l.gs.Status.Counters[name].Capacity}, nil
 	}
 	return nil, errors.Errorf("NOT_FOUND. %s Counter not found", name)
 }
 
+// GetList returns a List. Returns NOT_FOUND if the List does not exist.
+// [Stage:Alpha]
+// [FeatureFlag:CountsAndLists]
+func (l *LocalSDKServer) GetList(ctx context.Context, in *alpha.GetListRequest) (*alpha.List, error) {
+	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		return nil, errors.Errorf("%s not enabled", runtime.FeatureCountsAndLists)
+	}
+
+	l.logger.WithField("name", in.Name).Info("Getting List")
+	l.recordRequest("getlist")
+	l.gsMutex.RLock()
+	defer l.gsMutex.RUnlock()
+
+	if list, ok := l.gs.Status.Lists[in.Name]; ok {
+		return &alpha.List{Name: in.Name, Capacity: list.Capacity, Values: list.Values}, nil
+	}
+	return nil, errors.Errorf("NOT_FOUND. %s List not found", in.Name)
+}
+
+// UpdateList returns the updated List. Returns NOT_FOUND if the List does not exist (name cannot be updated).
+// **THIS WILL OVERWRITE ALL EXISTING LIST.VALUES WITH ANY REQUEST LIST.VALUES**
+// Use AddListValue() or RemoveListValue() for modifying the List.Values field.
+// Returns INVALID_ARGUMENT if the field mask path(s) are not field(s) of the List.
+// If a field mask path(s) is specified, but the value is not set in the request List object,
+// then the default value for the variable will be set (i.e. 0 for "capacity", empty list for "values").
+// [Stage:Alpha]
+// [FeatureFlag:CountsAndLists]
+func (l *LocalSDKServer) UpdateList(ctx context.Context, in *alpha.UpdateListRequest) (*alpha.List, error) {
```

Comment:
Do we also need to add validation logic to check the required fields in the request actually exist?

Reply:
At this point the fields do exist, because the default fields are added to the object at creation if not specified.

Reply:
Indeed, but aren't the default value for `List` and `UpdateMask` in the `UpdateListRequest` `nil`?

Reply:
Ah gotcha, I can add a nil check on the UpdateListRequest.